tools/python/mwm_downloader.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import argparse
+import os
+import re
+import time
+import sys
+import urllib.request
+import logging
+
+from concurrent.futures import ThreadPoolExecutor, as_completed
+
+
+DIRECT_MAP_URL = 'http://direct.mapswithme.com/direct/'
+ROOT = os.path.dirname(os.path.realpath(__file__))
+DEFAULT_DOWNLOAD_DIRECTORY = os.path.join(os.path.dirname(os.path.dirname(ROOT)), 'data')
+MAP_REVISION_REGEXP = re.compile(r'(\d{6})')
+MWM_NAME_REGEXP = re.compile(r'"(\S+).mwm"')
+
+logger = logging.getLogger(__name__)
+
+
+def parse_html_by_regexp(url, regexp):
+    parse_result = []
+    with urllib.request.urlopen(url) as f:
+        lines = f.read().decode('utf-8').split('\n')
+        for line in lines:
+            match = regexp.search(line)
+            if match:
+                parse_result.append(match.group(1))
+    return parse_result
+
+
+def get_possible_mwm_revisions():
+    return [int(revision) for revision in parse_html_by_regexp(DIRECT_MAP_URL, MAP_REVISION_REGEXP)]
+
+
+def get_possible_mwm_names(revision_num):
+    revision_url = '{base}{revision}'.format(base=DIRECT_MAP_URL, revision=revision_num)
+    return [name.replace('%20', ' ') for name in parse_html_by_regexp(revision_url, MWM_NAME_REGEXP)]
+
+
+def download_file(url, filename, attempts=3):
+    try:
+        urllib.request.urlretrieve(url, filename=filename)
+    except:
+        logger.error('Map with name {name} not loaded'.format(name=os.path.basename(filename)), exc_info=True)
```

Comment:
Please log an error

Reply:
Flag `exc_info=True` provide including traceback with error message in log